tests/unit/test_session.py

```diff
@@ -379,6 +379,13 @@ def test_provides_empty_list_for_unknown_service_regions(self):
         regions = self.session.get_available_regions('__foo__')
         self.assertEqual([], regions)
 
+    def test_provides_correct_partition_for_region(self):
```

Comment:
Can we get a couple more tests. I think we want to make sure we're validating the regex with something like `af-south-99` (doesn't exist but matches regex), and a partition other than `aws` (`cn-north-1` or `us-gov-east-1` would work).